backend/pkg/modules/internal/traceanalyzer/nlid/nlid.go

```diff
@@ -0,0 +1,280 @@
+// Copyright © 2022 Cisco Systems, Inc. and its affiliates.
+// All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package nlid
+
+import (
+	"bytes"
+	"container/ring"
+	"encoding/json"
+	"fmt"
+	"net/url"
+	"sort"
+	"strings"
+
+	log "github.com/sirupsen/logrus"
+
+	"github.com/openclarity/apiclarity/backend/pkg/modules/internal/core"
+	"github.com/openclarity/apiclarity/backend/pkg/modules/internal/traceanalyzer/utils"
+	pluginsmodels "github.com/openclarity/apiclarity/plugins/api/server/models"
+)
+
+const (
+	NLIDRingBufferSize = 1024
+)
+
+const (
+	MinIDValueLength = 8
+	MaxIDValueLength = 40 // This is at least a UUID. every string larger that than is considered low chance of being an ID
+)
+
+var IDKeys = [...]string{"id", "ids", "identifier", "identifiers"}
+
+type params = map[string]bool
+
+type Reason map[string]interface{}
```

Comment:
I don't see where it is being used

Reply:
it is on https://github.com/openclarity/apiclarity/blob/3f7fb314332913541f57c1185ac973ab2aecff7c/backend/pkg/modules/internal/traceanalyzer/nlid/nlid.go#L252

Reply:
It's used in traceanalyzer.go